File: frontend/src/objects/order/core/OrderPriceBreakdown.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { CurrencyCents } from '@/objects/core/SharedObjects'

export type OrderPriceBreakdown = {
  itemSubtotalCents: CurrencyCents
  couponDiscountCents: CurrencyCents
  totalPriceCents: CurrencyCents
}

File: backend/src/main/scala/services/order/objects/OrderPriceBreakdown.scala
package services.order.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.*

final case class OrderPriceBreakdown(
    itemSubtotalCents: CurrencyCents,
    couponDiscountCents: CurrencyCents,
    totalPriceCents: CurrencyCents,
)
